src/main/java/ua/mateacademy/tania/lesson17/jdbc/OfficeDaoImpl.java

```diff
@@ -45,6 +42,70 @@
         return offices;
     }
 
+    @Override
+    public Office getById(BigInteger id) throws SQLException {
+        Connection connection = ConnectionDB.getConnection();
+        PreparedStatement preparedStatement = connection.prepareStatement("SELECT* FROM offices WHERE office=?");
+        preparedStatement.setBigDecimal(1, new BigDecimal(id));
+        ResultSet rs = preparedStatement.executeQuery();
+
+        Office office = null;
+        if (rs.next()) {
+            office = extraxtFromResultSet(rs);
+        }
+
+        rs.close();
```

Comment:
Close it in finally block. If you have some exception this is don't work. In other also